File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

public interface BrandService {
    List<TbBrand> findAll();

    PageResult findByPage(Integer pageNum, Integer pageSize);

    void add(TbBrand brand);

    TbBrand findById(Long id);

    void update(TbBrand brand);

    void deleteBrands(Long... ids);

    PageResult findByPage(TbBrand brand, Integer pageNum, Integer pageSize);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务
app.service("brandService", function ($http) {
    this.findByPage = function (pageNum, pageSize) {
        return $http.get("../brand/findByPage?pageNum=" + pageNum + "&pageSize=" + pageSize);
    };

    this.add = function (brand) {
        return $http.post("../brand/add", brand);
    };

    this.update = function (brand) {
        return $http.post("../brand/update", brand);
    };

    this.findById = function (id) {
        return $http.get("../brand/findById?id=" + id);
    };

    this.deleteSelectBrands = function (ids) {
        return $http.get("../brand/deleteBrands?ids=" + ids);
    };

    this.search = function (pageNum, pageSize, brandSelect) {
        return $http.post("../brand/search?pageNum=" + pageNum + "&pageSize=" + pageSize, brandSelect);
    };

    this.findIdAndNameByBrand = function () {
        return $http.get("../brand/findIdAndName");
    }
});